### PR TITLE
[6.x] DS-672: Upgrade to webpack 5

### PR DESCRIPTION
**Pull request by danielamorse**
## Jira

https://pegadigitalit.atlassian.net/browse/DS-672

## Summary

Upgrade Webpack to v5, cleanup old and unused code, update browserslist config.

## Details

- Upgrade from Webpack v4 to v5 in both Bolt build and PL build
- Update Webpack configuration
- Update dependencies for compatibility
- Fix CSS throwing errors in upgraded SCSS loader
- Resolve dependency mismatch warnings
- Update browserslist to match our actual browser support strategy
- Remove unused dependencies
- Remove unused packages:
  - packages/website-ui/uikit-critical-css-webpack-plugin/**
  - packages/website-ui/uikit-prerenderer/**
  - packages/website-ui/uikit-prerender-spa-plugin/**
- Remove unused features:
  - criticalcss
- Remove dead build-related code

## How to test

- Review code
- Verify build, watch, test commands all work as expected
- Review a variety of pages in PL for regressions

## Release notes
- Webpack in `@bolt/build-tools` has been upgraded from v4 to v5.
- Browserslist configuration in `@bolt/browserslist-config` has been updated.

### CODE REVIEW

File: packages/build-tools/create-webpack-config.js [hunk 1]
```diff
@@ -312,22 +266,18 @@ async function createWebpackConfig(buildConfig) {
         },
       }),
       new webpack.ProgressPlugin(boltWebpackProgress), // Ties together the Bolt custom Webpack messages + % complete
-      new webpack.NoEmitOnErrorsPlugin(),
+      new webpack.NoEmitOnErrorsPlugin(), // ?
```
Comment (colbytcook) [inline packages/build-tools/create-webpack-config.js:269]: What a powerful comment...

File: packages/build-tools/utils/formatWebpackMessages.js [hunk 2]
```diff
@@ -26,7 +26,7 @@ function isLikelyASyntaxError(message) {
 
 // Cleans up webpack error messages.
 // eslint-disable-next-line no-unused-vars
```
Comment (colbytcook) [inline packages/build-tools/utils/formatWebpackMessages.js:28]: Do we still need this Eslint disable comment?

**Review by colbytcook — APPROVED** [review]
@danielamorse great work, I love seeing all the code removal. The code looks good and I can confirm the following commands operate correctly.
- `yarn setup`
- `yarn build`
- `yarn`
- `yarn start`
- `yarn test:js`
- `yarn test:js -u`

File: packages/build-tools/create-webpack-config.js (hunk 1 above)
Comment (danielamorse) [reply packages/build-tools/create-webpack-config.js:269]: lol, that was a "note to self" that slipped through the cracks. This plugin is removed in the bundle-analyzer PR 😆

File: packages/build-tools/utils/formatWebpackMessages.js (hunk 2 above)
Comment (danielamorse) [reply packages/build-tools/utils/formatWebpackMessages.js]: No, I don't see why we'd need that. Nice catch, removed.

**Review by adamszalapski — APPROVED** [review]
Two things that are conspicuous are:
1. How fast `yarn setup` works -> 41.04s
2. How fast `yarn start` work -> 33.369s
That are fantastic results. Thank you @danielamorse for that.
`yarn build` is longer but much faster that the previous one -> 73.11s
All commands work as expected including `yarn cc` and `yarn ce`
I noticed that after running `yarn start` and working with the code we can see that `app` is undefined and I don't know if this is expected or not.
image

**Pull request merged**